### ISSUE TITLE
Feature request: display string lengths in var dumper

### ISSUE THREAD

**Issue by Alorel** [symfony/symfony#15856]
Hey folks. I've just started using the var dumper and it seems to lack a rather useful feature: displaying string lengths. 

## How it currently works ##
If one of the variables passed to the dumper is a string, it will be dumped as `"string-contents"`

    $var = ['foo'];
    VarDumper::dump($var);
    >>>
    array:1 [
       0 => & "foo"
    ]

## Proposed change ##
Display the string length in a manner similar to how array lengths are displayed:

    $var = ['foo'];
    VarDumper::dump($var);
    >>>
    array:1 [
       0 => & string:3 "foo"
    ]

A working example can be found in [Raveren's Kint](https://github.com/raveren/kint) project.

**Comment by iltar** [comment]
The double quotes already imply that it's a string, what about a more intuitive syntax?

```
array:1 [
   0 => & 3:"foo"
   0 => & "foo" (3)
   0 => & (3) "foo"
]

```

**Comment by Alorel** [comment]
@iltar Indeed, my proposed syntax was a bit too much - the core proposal is to display the string *length* in the output, not the fact that it's a string.

**Comment by nicolas-grekas** [comment]
fyi, length is displayed on hover in html mode, so you'd like it in cli output also?

**Comment by nicolas-grekas** [comment]
(btw, my preference would go for `"foo" (3)`)

**Comment by Alorel** [comment]
@nicolas-grekas I never noticed the title attribute in html - thanks, but yes, something similar would be highly appreciated in cli mode (my preference would be the same). As for html output, perhaps there can be an option to toggle display as `"foo" (3)` instead of hardcoding it as a title attribute (as it is not immediately obvious at this stage)?

**Comment by iltar** [comment]
The downside of post-fixing it, means you won't be able to easily see the length without checking the end. With long strings this might be a bit annoying.

**Comment by nicolas-grekas** [comment]
Putting (what I think is) a secondary information in front of the main information would be just clutter readability. Personally, I never read the length. It's usually useless information to me. That's why the output is the way it is now. IMHO, the readability loss is higher than the gain. I'd really like to discuss the need to output the length before we implement it if you think it's worth.

**Comment by iltar** [comment]
One case where you might need it, is comparing strings that can contain zero-width spaces for example.

**Comment by nicolas-grekas** [comment]
I'm sure there is a use case, but should we clutter all and every dump for just in case someone is in this very rare situation?

**Comment by Alorel** [comment]
Another (simple) use case is when you're dealing wish hashes - if you see an invalid length you know you missed something in your code.

A third use case is dealing with encrypted data in limited-length datastores, allowing you to estimate how long the current code implementation can be used for.

I see your point about clutter - indeed, there's no need to force this upon all users, so it *could* be made as an optional setting which is disabled by default.

**Comment by nicolas-grekas** [comment]
See #17305

**Comment by javiereguiluz** [comment]
I'm closing this as fixed because the feature was implemented by Nicolas in #17305 via the `DUMP_STRING_LENGTH` flag.
